compiler/il/OMRNode.cpp

```diff
@@ -3359,6 +3359,42 @@ OMR::Node::getOwningMethod()
    }
 
 
+/** \brief
+ *  Used to get the ram method from the Bytecode Info.
+ * 
+ *  \param *comp
+ *  _compilation of type TR::Compilation
+ * 
+ *  \param &bcInfo 
+ *  _byteCodeInfo of a node of type TR::Node  
+ *  
+ *  @return 
+ *  Returns the ram method (TR_OpaqueMethodBlock)
+ */
+TR_OpaqueMethodBlock* 
+OMR::Node::getOwningMethod(TR::Compilation *comp, TR_ByteCodeInfo &bcInfo)
```

Comment:
Why is this an instance method if you don't use the node? Wouldn't it make sense for the Node to have a getOwningMethod that used the Node's bcInfo and that there is a static method that takes a bcInfo so you can ask the question without having a node and the node implementation delegates to it?

Reply:
@andrewcraik Could you please clarify why is this called as an instance method when it is declared as static? This routine is called without an object (node).

Reply:
Sorry I missed the static in the hpp - it being static makes sense, but having the instance method to take the node's bcinfo and pass it to the static function will simplify usage since most of the bcinfo will come from a node.

Reply:
Initially, there were two functions, I have removed one of them, realizing this. Now there's only one static routine which takes comp and bcinfo as inputs. Please let me know if I understood it right.

Reply:
No I'm suggesting an instance method on Node called getOwningMethod which only takes a comp and passes the Node's own bcInfo to the static version of getOwningMethod. I had a look at the usage in OpenJ9 in your proposed PR and there were a number of places where we have a node and call getOwningMethod which results in the code getting messier with the getByteCodeInfo being added. With those two versions everything else can then clean up IMO.

Reply:
The bcInfo is used to called other functions like getByteCodeIndex(), doNotProfile() in the same scope, except in one or two cases, where node->getByteCodeInfo() is used (in my latest commit in OpenJ9). In some other cases, there is only bcInfo and no node. I don't think I can cleanup bcInfo if I have an other instance method. Please let me know if you'd want me to change anything.

Reply:
I'm not sure I see the clean-up problem you are facing by having an:
```
TR_OpaqueMethodBlock*
OMR::Node::getOwningMethod(TR::Compilatin *comp) {
   return getOwningMethod(comp, self()->getByteCodeInfo());
}
```

In addition to the static `TR_OpaqueMethodBlock* OMR::Node::getOwningMethod(TR::Compilation *comp, TR_ByteCodeInfo &bcInfo) { ... }`